packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -378,19 +378,32 @@ abstract class WidgetsBinding extends BindingBase with GestureBinding, RendererB
   }
 
   bool _needToReportFirstFrame = true;
-  bool _thisFrameWasUseful = true;
+  int _deferFirstFrameReportCount = 0;
+  bool get _reportFirstFrame => _deferFirstFrameReportCount == 0;
 
-  /// Tell the framework that the frame we are currently building
-  /// should not be considered to be a useful first frame.
+  /// Tell the framework not to report the frame it is building as a "useful"
+  /// first frame until there is a corresponding call to [allowFirstFrameReport].
   ///
   /// This is used by [WidgetsApp] to report the first frame.
   //
   // TODO(ianh): This method should only be available in debug and profile modes.
-  void preventThisFrameFromBeingReportedAsFirstFrame() {
-    _thisFrameWasUseful = false;
+  void deferFirstFrameReport() {
+    assert(_deferFirstFrameReportCount >= 0);
+    _deferFirstFrameReportCount += 1;
   }
 
-  void _handleBuildScheduled() {
+  /// Tell the framework to report the frame it is building as a "useful"
+  /// first frame.
+  ///
+  /// This is used by [WidgetsApp] to report the first frame.
```

Comment:
this documentation is wrong. It makes it sound like you should call this even if you haven't called defer.